pkg/mimir/modules.go

```diff
@@ -913,19 +914,35 @@ func (t *Mimir) initQueryFrontend() (serv services.Service, err error) {
 	}), nil
 }
 
-func (t *Mimir) initQueryPlanner() (services.Service, error) {
-	_, mqeOpts := engine.NewPromQLEngineOptions(t.Cfg.Querier.EngineConfig, t.ActivityTracker, util_log.Logger, t.Registerer)
-	t.QueryPlanner = streamingpromql.NewQueryPlanner(mqeOpts)
+func (t *Mimir) initQuerierQueryPlanner() (services.Service, error) {
+	reg := prometheus.WrapRegistererWith(prometheus.Labels{"component": "querier"}, t.Registerer)
+	_, mqeOpts := engine.NewPromQLEngineOptions(t.Cfg.Querier.EngineConfig, t.ActivityTracker, util_log.Logger, reg)
+	t.QuerierQueryPlanner = streamingpromql.NewQueryPlanner(mqeOpts)
 
-	analysisHandler := streamingpromql.AnalysisHandler(t.QueryPlanner)
+	// If the query-frontend is running in this process, we want to expose its planner through the analysis endpoint, not
+	// the querier's planner.
+	if !t.Cfg.isQueryFrontendEnabled() {
```

Comment:
The conditional in the comment is technically correct but the opposite of what the code is doing. Maybe "Only expose the querier's planner when the query-frontend is not running in this process"